### PR TITLE
Issue-43-39 Fixed issue with exceptions being swallowed in a threaded…

### LINKED ISSUE #39: No error if I use an invalid command file
I ran this:
```python
from dragoneye import AwsScanner, AwsCloudScanSettings, AzureScanner, AzureCloudScanSettings, AwsSessionFactory, AzureAuthorizer

aws_settings = AwsCloudScanSettings(
    commands_path='/Users/dev/python/dragoneye/aws_commands_example.yaml',
    account_name='default',
    default_region='us-east-1',
    regions_filter=['us-east-1']
)

session = AwsSessionFactory.get_session_using_assume_role(external_id='<redacted>',
                                                          role_arn="<redacted>",
                                                          region='us-east-1')
aws_scan_output_directory = AwsScanner(session, aws_settings).scan()
```

And got:
```
* Getting region names
* Creating directory for each region name
--------------------------------------------------------------------
Summary: 0 APIs called. 0 errors
Scanning AWS live environment took 0.710157 seconds
```

It didn't tell me that I provided an invalid path for the command file.

### LINKED ISSUE #43: Threaded functions swallows exceptions
All functions that are running in a separate thread swallow the exceptions without any indication of a failure

### PR DESCRIPTION
**Pull request by Tomerva**
fix #43 fix #39

### CODE REVIEW

**Comment by Viperain** [comment]
@Tomerva can you describe what was the original behavior and how you changed it ? for example, an exception in the thread would not fail the entire collect? is it going to fail it now?

**Comment by Tomerva** [comment]
> @Tomerva can you describe what was the original behavior and how you changed it ? for example, an exception in the thread would not fail the entire collect? is it going to fail it now?

All the exceptions that happened in a thread were ignored.
Now, exceptions are caught by their caller and prints an error message describing the exception.
Any exception in the threads did not, and will not, fail the process. The difference is that now we are showing it.

File: dragoneye/cloud_scanner/azure/azure_scanner.py [hunk 1]
```diff
@@ -36,29 +37,42 @@ def scan(self) -> str:
...
-        request = scan_command['Request']
-        parameters = scan_command.get('Parameters', [])
-        url = request.replace('{subscriptionId}', subscription_id)
-        results = AzureScanner._get_results(url, headers, parameters, account_data_dir, resource_groups)
-        self._save_result(results, output_file)
+        try:
+            output_file = self._get_result_file_path(account_data_dir, scan_command['Name'])
+            if os.path.isfile(output_file):
+                # Data already scanned, so skip
+                logger.warning("  Response already present at {}".format(output_file))
```
Comment (Viperain) [inline dragoneye/cloud_scanner/azure/azure_scanner.py:66]: remove space and use '

**Review by Viperain — APPROVED** [review]
one small comment

**Pull request merged**